pkg/blockfactory/blockissuer.go

```diff
@@ -163,24 +167,53 @@ func (i *BlockIssuer) CreateValidationBlock(ctx context.Context, opts ...options
 	return modelBlock, nil
 }
 
+func (i *BlockIssuer) getCommitment(blockSlot iotago.SlotIndex) (*iotago.Commitment, error) {
+	protoParams := i.protocol.CurrentAPI().ProtocolParameters()
+	commitment := i.protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment()
+
+	if blockSlot > commitment.Index+protoParams.MaxCommittableAge() {
+		return nil, ierrors.Errorf("can't issue block: block slot %d is too far in the future, latest commitment is %d", blockSlot, commitment.Index)
+	}
+
+	if blockSlot < commitment.Index+protoParams.MinCommittableAge() {
+		if blockSlot < protoParams.MinCommittableAge() || commitment.Index < protoParams.MinCommittableAge() {
```

Comment:
As `blockSlot` is provided by the user it can be any slot. We only provide facility to create a _syntactically valid_ block here. It doesn't mean that it's a going to be _useful_ block. Therefore, we make sure that if `blockSlot` is smaller than `commitment.Index` that we either return said commitment (in case it's too close to 0) or retrieve a correct one for this `blockSlot` later on